protocol/contracts/libraries/LibFertilizer.sol

```diff
@@ -230,4 +229,33 @@ library LibFertilizer {
         LibUnripe.decrementUnderlying(C.UNRIPE_LP, balanceOfUnderlying);
         LibUnripe.switchUnderlyingToken(C.UNRIPE_LP, well);
     }
+
+    /**
+     * Taken from previous versions of OpenZeppelin Library.
+     * @dev Returns true if `account` is a contract.
+     *
+     * [IMPORTANT]
+     * ====
+     * It is unsafe to assume that an address for which this function returns
+     * false is an externally-owned account (EOA) and not a contract.
+     *
+     * Among others, `isContract` will return false for the following
+     * types of addresses:
+     *
+     *  - an externally-owned account
+     *  - a contract in construction
+     *  - an address where a contract will be created
+     *  - an address where a contract lived, but was destroyed
+     * ====
+     */
+    function isContract(address account) internal view returns (bool) {
```

Comment:
Believe in solidity 0.8.0 and up, the evm now exposes the `code` member, which allows you to determine whether the address is a contract. https://docs.soliditylang.org/en/latest/units-and-global-variables.html?highlight=address+initial#members-of-address-types. This definitely works as well but defer to you on implmentation.

Reply:
I didn't want to shake the tree on the intricacies of how this vs .code might behave differently during construction or after destruction. current open zeppelin version does not offer any form of contract checking.